Add a `WordPressLoggingDelegate` implementation that logs to console

mokagio opened the pull request:
When working on the WordPress Authenticator  demo app for Google SignIn testing, I got a message about `WordPressLoggingDelegate` not being implemented. I put together this simpler implementation to `print`s to the console to satisfy the requirement. I think it might be useful during development or debugging and worth shipping together with the `protocol`.


---

- [x] I have considered if this change warrants release notes and have added them to the appropriate section in the `CHANGELOG.md` if necessary.

mokagio committed:
Add a `WordPressLoggingDelegate` implementation that logs to console

Useful during development or debugging.

mokagio committed:
Add `CHANGELOG.md` entry about `ConsoleLogger`

crazytonyli reviewed (approved):
Love the emojis :shipit: 

BTW, I feel like `NSLog` would give us more (sometimes quite important) info in the logs, like timestamp, thread id, etc. My personal preference is `NSLog` over `print`, but I don't really have a strong opinion on this.

mokagio committed:
Use `NSLog` instead of `print` for richer logs

See suggestion at
https://github.com/wordpress-mobile/WordPress-iOS-Shared/pull/335#pullrequestreview-1274296989

`print` logs:

> 📃 – Verbose: Received Social Login Oauth response.

`NSLog` logs:

> 2023-01-30 14:37:07.251471+1100 AuthenticatorDemo[47512:28088466] 📃 – Verbose: Received Social Login Oauth response.

mokagio commented:
Thank you for the `NSLog` suggestion @crazytonyli. It does provide more info indeed, at no extra cost:

```
2023-01-30 14:37:07.251471+1100 AuthenticatorDemo[47512:28088466] 📃 – Verbose: Received Social Login Oauth response.
```

vs 

```
📃 – Verbose: Received Social Login Oauth response.
```

Implemented in [wordpress-mobile/WordPress-iOS-Shared@`cd428c3` (#335)](https://github.com/wordpress-mobile/WordPress-iOS-Shared/pull/335/commits/cd428c3fcda83b36977eaffc795767ed358c43d6)

mokagio commented:
By the way @crazytonyli, I wanted to run the tests locally to see if my changes compiled but I got the following error from Fastlane:

```
xcodebuild: error: The workspace named "WordPressShared" does not contain a scheme named "WordPressShared". The "-list" option can be used to find the names of the schemes in the workspace.
```

I wonder if you get a similar behavior on your end, but I suspect there must be something wrong on my side as CI wouldn't be green otherwise.

crazytonyli commented:
@mokagio It's likely caused by the left over xcworkspace or xcodeproj file, which were deleted after adding SPM support. If those files are still there, the `xcodebuild` command would pick them up as the project to build, instead of the `Package.swift`.

mokagio commented:
Spot on. I should have realized 🤦‍♂️ Thanks @crazytonyli !
